internal/util/setters/setters.go

```diff
@@ -29,30 +31,53 @@ import (
 	"sigs.k8s.io/kustomize/kyaml/openapi"
 	"sigs.k8s.io/kustomize/kyaml/pathutil"
 	"sigs.k8s.io/kustomize/kyaml/setters"
+	"sigs.k8s.io/kustomize/kyaml/setters2"
 	"sigs.k8s.io/kustomize/kyaml/setters2/settersutil"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
 )
 
 const (
 	GcloudProject       = "gcloud.core.project"
 	GcloudProjectNumber = "gcloud.project.projectNumber"
 )
 
-func PerformAutoSetters(path string) error {
+type AutoSet struct {
+	Writer     io.Writer
+	TargetPath string
```

Comment:
Can we clarify the difference between TargetPath and ParentPath?

Reply:
Updated the PR. Renamed TargetPath to PackagePath and added comment. Removed ParentPath parameter as it is always the parent directory of PackagePath.